lib/CMakeLists.txt

```diff
@@ -311,20 +311,39 @@ bison_target(ParserExprGrammar
     COMPILE_FLAGS ${BISON_FLAGS})
 add_flex_bison_dependency(CfgLexer ParserExprGrammar)
 
-set(CORE_INCLUDE_DIRS
-    ${Gettext_INCLUDE_DIR}
-    ${IVYKIS_INCLUDE_DIR}
-    ${LIBPCRE_INCLUDE_DIRS}
-    ${Libsystemd_INCLUDE_DIRS}
-)
+if(NOT IVYKIS_INTERNAL)
+    set(CORE_INCLUDE_DIRS
+        ${Gettext_INCLUDE_DIR}
+        ${IVYKIS_INCLUDE_DIR}
+        ${LIBPCRE_INCLUDE_DIRS}
+        ${Libsystemd_INCLUDE_DIRS}
+    )
+else()
+    set(CORE_INCLUDE_DIRS
+        ${Gettext_INCLUDE_DIR}
+        ${LIBPCRE_INCLUDE_DIRS}
+        ${Libsystemd_INCLUDE_DIRS}
```

Comment:
I don't think this is a proper fix.
`IVYKIS_INCLUDE_DIR` is required in both "internal" and "system" cases, and should be set properly by `external_or_find_package`.

The commit message suggests that even with `--with-ivykis=internal`, system-provided ivykis libraries are included. Probably the source of the system-provided ivykis headers needs to be found and fixed there.